packages/sanes-chrome-extension/src/extension/background/actions/createPersona/requestHandler.ts

```diff
@@ -49,4 +49,22 @@ export class RequestHandler {
 
     return size;
   }
+
+  public static purge(senderUrl: string): void {
+    const initialSize = RequestHandler.instance.length;
+    for (let i = 0; i < initialSize; i++) {
+      const req = RequestHandler.instance[i];
+      if (req.sender !== senderUrl) {
+        continue;
+      }
+
+      // Note here we are mutating the array placing in first positions request to be rejected
+      RequestHandler.instance.splice(i, 1);
+      RequestHandler.instance.splice(0, 0, req);
+    }
```

Comment:
Hmm, if reject() would not automatically solve the first element, we would not need this reordering. Anyways, this seems to be correct

Reply:
Same as before, I will check again.